src/olympia/addons/tests/test_views.py

```diff
@@ -2731,6 +2731,41 @@ def test_filter_by_multiple_authors(self):
         assert result['id'] == addon2.pk
         assert result['slug'] == addon2.slug
 
+    def test_filter_by_guid(self):
+        addon = addon_factory(slug='my-addon', name=u'My Addôn',
+                              guid='random@guid', weekly_downloads=999)
+        addon_factory()
+        self.reindex(Addon)
+
+        data = self.perform_search(self.url, {'guid': u'random@guid'})
+        assert data['count'] == 1
+        assert len(data['results']) == 1
+
+        result = data['results'][0]
+        assert result['id'] == addon.pk
+        assert result['slug'] == addon.slug
+
+    def test_filter_by_multiple_guid(self):
+        addon = addon_factory(slug='my-addon', name=u'My Addôn',
+                              guid='random@guid', weekly_downloads=999)
+        addon2 = addon_factory(slug='another-addon', name=u'Another Addôn',
+                               guid='random2@guid',
+                               weekly_downloads=333)
+        addon_factory()
+        self.reindex(Addon)
+
+        data = self.perform_search(
+            self.url, {'guid': u'random@guid,random2@guid'})
+        assert data['count'] == 2
+        assert len(data['results']) == 2
+
+        result = data['results'][0]
+        assert result['id'] == addon.pk
+        assert result['slug'] == addon.slug
+        result = data['results'][1]
+        assert result['id'] == addon2.pk
+        assert result['slug'] == addon2.slug
```

Comment:
Can we add some random other guid string to the search string to test the behavior of invalid guids?

Reply:
What kind of invalid guid and to test what exactly?  If it's an invalid guid it won't be found, like any other guid we don't have in the database.

Reply:
Right, that means if you're adding a guid that won't be found to line 2757 the assertion in 2760 will still be `2` which is what I want to see basically

Reply:
We don't have a "not found" test for guids for that filter as far as I can tell